contracts/interfaces/IbcMiddleware.sol

```diff
@@ -247,3 +250,29 @@ contract IbcMwUser is Ownable {
         authorizedMws[address(middleware)] = true;
     }
 }
+
+abstract contract IbcUniversalPacketReceiverBase is IbcMwUser, IbcUniversalPacketReceiver {
+    mapping(bytes32 => bool) public authorizedChannelIds;
+
+    /**
+     * @dev To ensure that UCH packets are sent over the UCH channel, we also have to validate the channelId
+     * @dev For onRecvUniversalPacket, the packet.dest.channelId should be that of the UCH channel of the counterparty
+     * chain. For onUniversalAcknowledgement, the packet.src.channelId should be the UCH channel of the counterparty
+     * chain
+     */
+    modifier onlyAuthorizedChannel(bytes32 channelId) {
+        if (!authorizedChannelIds[channelId]) {
+            revert unauthorizedChannel();
+        }
+        _;
+    }
+
+    constructor(address _middleware) IbcMwUser(_middleware) {}
+
+    function _authorizeChannel(bytes32 channelId) internal {
```

Comment:
add detailed docs for what this channelID should be?
eg. it must be a UC channelID from the running chain's perspective, or better with an example, a UC with two ends, channel-1 on chain A and channel-2 on B, then contract on A should authorize channel-1, etc

Reply:
added some natspec with an example!

Reply:
I think this specific doc is relevant to issue 23 so I added it to [that commit ](https://github.com/open-ibc/vibc-core-smart-contracts/pull/108/commits/1da74e81d86989b537a2a032406c8e31050a70c3)